tower-reconnect/src/lib.rs

```diff
@@ -23,47 +24,47 @@ pub enum Error<T, U> {
     NotReady,
 }
 
-pub struct ResponseFuture<T, Request>
+pub struct ResponseFuture<T, Ctx, Request>
```

Comment:
I know it is an unrelated change... but can this just be `ResponseFuture<T>` where `T` is the inner future?

Reply:
Not easily, since the error is mapped by `Error::Inner` which means the bounds are needed. Perhaps with some better refactoring...